### ISSUE TITLE
The explorer context menu should only appear on shift+click

### ISSUE THREAD

**Issue by zadjii-msft** [microsoft/terminal#6113]
Follow-up from #6100. 

The Explorer context menu is notoriously cluttered. We shouldn't add to that clutter if we can avoid it. This issue represents me taking the time to investigate how exactly to make the menu only appear on shift+right click, not _just_ right click.

**Comment by zadjii-msft** [comment]
Alright so we took a quick peek through the code and we're not sure exactly how to do this. Fortunately, PowerToys does seem to do something similar:

https://github.com/microsoft/PowerToys/blob/d16ebba9e0f06e7a0d41d981aeb1fd0a78192dc0/src/modules/powerrename/dll/PowerRenameExt.cpp#L60-L110

Looks like they implement another interface, `IContextMenu`. If someone in the community was passionate about decluttering the context menu, we'd definitely appreciate a similar PR 😄

**Comment by rescenic** [comment]
Yeah, it's ugly long name. I am trying to disable that context menu. But I can't find Windows Terminal Preview context menu item using Nirsoft tools, like ShellExView and ShellMenuView.
![img](https://i.ibb.co/zZChTNN/23-06-2020-863-23-33-27.png)

**Comment by igorakkerman** [comment]
Please make this configurable, Terminal is a core application in my work and should not be a secondary function.
